### PR TITLE
Hp delete lb

### PR DESCRIPTION
**Pull request by houpeng80**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/cloudprovider/huaweicloud/loadbalancer.go [hunk 1]
```diff
@@ -710,6 +710,44 @@ func (l *LB) UpdateLoadBalancer(ctx context.Context, clusterName string, service
...
+	if err != nil {
+		return err
+	}
+	// query ELB listeners list
+	listenerArr, err := lbService.ListListeners(loadBalancer.ID)
+	if err != nil {
+		return err
+	}
+	err = l.deleteListeners(loadBalancer, listenerArr, &ensureOptions{lbServices: lbService})
+	if err != nil {
```
Comment (deer-hang) [inline pkg/cloudprovider/huaweicloud/loadbalancer.go]: if err := l.deleteListeners(loadBalancer, listenerArr, &ensureOptions{lbServices: lbService}); err != nil {
    return err
}
write in a line maybe better

File: pkg/cloudprovider/huaweicloud/loadbalancer.go [hunk 2]
```diff
@@ -710,6 +710,44 @@ func (l *LB) UpdateLoadBalancer(ctx context.Context, clusterName string, service
...
+		return err
+	}
+	err = l.publicIpService.UnbindAndDeleteEip(loadBalancer.VipPortID, params.eipID, params.keepEip)
+	if err != nil {
```
Comment (deer-hang) [inline pkg/cloudprovider/huaweicloud/loadbalancer.go]: write in a line maybe better.

File: pkg/cloudprovider/huaweicloud/loadbalancer.go [hunk 3]
```diff
@@ -710,6 +710,44 @@ func (l *LB) UpdateLoadBalancer(ctx context.Context, clusterName string, service
...
+		return err
+	}
+	err = lbService.Delete(loadBalancer.ID)
+	if err != nil {
```
Comment (deer-hang) [inline pkg/cloudprovider/huaweicloud/loadbalancer.go]: write in a line maybe better.

File: pkg/cloudprovider/huaweicloud/loadbalancer.go (hunk 1 above)
Comment (houpeng80) [reply pkg/cloudprovider/huaweicloud/loadbalancer.go]: ok,it's a good idea

Comment (houpeng80) [reply pkg/cloudprovider/huaweicloud/loadbalancer.go]: ok

File: pkg/cloudprovider/huaweicloud/loadbalancer.go (hunk 2 above)
Comment (houpeng80) [reply pkg/cloudprovider/huaweicloud/loadbalancer.go]: ok

**Pull request closed without merging**